Go back to F37 to fix kokkos CI

> Looks like this broke our CI again, see https://github.com/kokkos/kokkos/actions/runs/5661994926/job/15341048991?pr=6200, 

```
CMake Error at cmake/kokkos_test_cxx_std.cmake:91 (MESSAGE):
  C++17-compliant compiler detected, but unable to compile C++17 or later
  program.  Verify that Intel:2021.10.0.20230609 is set up correctly (e.g.,
  check that correct library headers are being used).

  Failing output:

   Change Dir: '/__w/kokkos/kokkos/builddir/corner_cases/CMakeFiles/CMakeTmp'

  

  Run Build Command(s): /usr/bin/cmake -E env VERBOSE=1 /usr/bin/gmake -f
  Makefile cmTC_4c411/fast

  /usr/bin/gmake -f CMakeFiles/cmTC_4c411.dir/build.make
  CMakeFiles/cmTC_4c411.dir/build

  gmake[1]: Entering directory
  '/__w/kokkos/kokkos/builddir/corner_cases/CMakeFiles/CMakeTmp'

  Building CXX object CMakeFiles/cmTC_4c411.dir/cplusplus17.cpp.o

  /opt/intel/oneapi/compiler/latest/linux/bin/intel64/icpc -Werror
  -diag-disable=177,10441 -std=gnu++17 -MD -MT
  CMakeFiles/cmTC_4c411.dir/cplusplus17.cpp.o -MF
  CMakeFiles/cmTC_4c411.dir/cplusplus17.cpp.o.d -o
  CMakeFiles/cmTC_4c411.dir/cplusplus17.cpp.o -c
  /__w/kokkos/kokkos/cmake/compile_tests/cplusplus17.cpp

  In file included from
  /__w/kokkos/kokkos/cmake/compile_tests/cplusplus17.cpp(17):

  /usr/include/c++/13/type_traits(3363): error: type name is not allowed

      inline constexpr bool is_convertible_v = __is_convertible(_From, _To);
                                                                ^

  

  In file included from
  /__w/kokkos/kokkos/cmake/compile_tests/cplusplus17.cpp(17):

  /usr/include/c++/13/type_traits(3363): error: type name is not allowed

      inline constexpr bool is_convertible_v = __is_convertible(_From, _To);
                                                                       ^

  

  In file included from
  /__w/kokkos/kokkos/cmake/compile_tests/cplusplus17.cpp(17):

  /usr/include/c++/13/type_traits(3363): error: identifier "__is_convertible"
  is undefined

      inline constexpr bool is_convertible_v = __is_convertible(_From, _To);
                                               ^

  

  compilation aborted for
  /__w/kokkos/kokkos/cmake/compile_tests/cplusplus17.cpp (code 2)

  gmake[1]: *** [CMakeFiles/cmTC_4c411.dir/build.make:79:
  CMakeFiles/cmTC_4c411.dir/cplusplus17.cpp.o] Error 2

  gmake[1]: Leaving directory
  '/__w/kokkos/kokkos/builddir/corner_cases/CMakeFiles/CMakeTmp'

  gmake: *** [Makefile:127: cmTC_4c411/fast] Error 2



Call Stack (most recent call first):
  cmake/kokkos_tribits.cmake:202 (INCLUDE)
  CMakeLists.txt:226 (KOKKOS_SETUP_BUILD_ENVIRONMENT)


-- Configuring incomplete, errors occurred!
Error: Process completed with exit code 1.
```

_Originally posted by @masterleinad in https://github.com/kokkos/ci-containers/issues/39#issuecomment-1650610408_

--- a/.github/workflows/continuous-integration-workflow.yml
+++ b/.github/workflows/continuous-integration-workflow.yml
@@ -21,7 +21,7 @@ jobs:
           - {dockerfile: 'ubuntu',   tag: 'rolling',          build_args: 'TAG=rolling'}
           - {dockerfile: 'ubuntu',   tag: 'devel',            build_args: 'TAG=devel', continue-on-error: 'true'}
           - {dockerfile: 'opensuse', tag: 'latest'}
-          - {dockerfile: 'fedora',   tag: 'intel',            build_args: 'INTEL=yes', continue-on-error: 'true'}
+          - {dockerfile: 'fedora',   tag: 'intel',            build_args: 'TAG=37,INTEL=yes', continue-on-error: 'true'}
     runs-on: ubuntu-latest
     steps:
       - name: Checkout out code

